pkg/cvo/upgradeable.go

```diff
@@ -442,3 +449,31 @@ func (optr *Operator) adminGatesEventHandler() cache.ResourceEventHandler {
 		DeleteFunc: optr.deleteFunc,
 	}
 }
+
+// shouldSyncUpgradeableDueToPreconditionChecks checks if the upgradeable status should
+// be synchronized due to the precondition checks. It checks whether the precondition
+// checks on the payload are failing for less than minimumUpdateCheckInterval, and it has
+// been more than the statusInterval since the last synchronization. This means we will
+// synchronize the upgradeable status more frequently at beginning of an upgrade for
+// a while in a case the precondition checks are failing.
+//
+// shouldSyncUpgradeableDueToPreconditionChecks expects the parameters not to be nil.
+//
+// Function returns true if the synchronization should happen, returns false otherwise.
+//
+// A precondition check preventing an upgrade could have been potentially already resolved
+// (e.g. operator reporting Upgradeable==false just momentarily).
+// CVO should synchronize the upgradeable status again in case of failing precondition
+// checks to speed up initializing an upgrade stuck on a precondition check that
+// potentially has been solved.
```

Comment:
Not sure this is needed for this function's godoc but if so I would try and combine with the above and make it all more concise or it may not be read.

Reply:
Ah, yes, I agree. I'll try to improve the comments adequately for the new commit.